sunpy/map/maputils.py

```diff
@@ -124,6 +124,17 @@ def sample_at_coords(smap, coordinates):
     return smap.data[smap.wcs.world_to_array_index(coordinates)]
 
 
+def _edge_coordinates(smap):
+    # We need to strip the units from edges before handing it to np.concatenate,
+    # as the .unit attribute is not propagated in np.concatenate for numpy<1.17
+    # When sunpy depends on numpy>=1.17 this unit replacing code can be removed
+    edges = [edge.to_value(u.pix) for edge in map_edges(smap)]
+    # Calculate all the edge pixels
+    edge_pixels = u.Quantity(np.concatenate(edges), unit=u.pix, copy=False)
```

Comment:
If performance is the driving concern here, why strip the unit only to put it back, even when such an operation is not needed (i.e., newer versions of NumPy)?  Especially a list comprehension!  Maybe the performance hit is negligible in actuality, but it looks silly.

How about using a private function:
```suggestion
    edge_pixels = np.concatenate(map_edges(smap))
    # The .unit attribute is not propagated in np.concatenate for numpy<1.17
    # When sunpy depends on numpy>=1.17 this unit replacing code can be removed
    if edge_pixels.unit != u.pix:
        edge_pixels._set_unit(u.pix)
```

Reply:
I think the performance hit is tiny, there's only 4 items in `map_edges(smap)`, and as profiled the bottleneck is now (re-)computing the WCS. In addition:

- I (personally) refuse to use private API from another library
- I don't think the code snippet would work because the `.unit` attribute doesn't exist for older versions of astropy

Reply:
Do you actually need to strip the units? Can't you just run whatever the output of `np.concatenate` is through `u.Quantity` and let that fix it?

Reply:
I originally did that but changed it in response to https://github.com/sunpy/sunpy/pull/4463#discussion_r491718451

Reply:
> * I don't think the code snippet would work because the `.unit` attribute on `edge_pixels` doesn't exist for older versions of numpy after going through `np.concatenate`

Well, that's broken then.  `.unit` is a property, so that means that `._unit` is `None`.  You could check for `None` instead, and I think the use of private API is justifiable in this case.